module-2/backend/week_four.md

```diff
@@ -10,14 +10,35 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+>stores information ofof the current session on the browser/app
+
 * What’s the difference between a session and a cookie?
+>Session is encrypted
+
 * What’s a flash and when do you want to use flashes?
+>a message that lets you know when you've done a thing.
+
 * Why do people say “HTTP is stateless”?
+>Because the after the response ,the connection is closed and therefore no state.
+
 * What’s authentication? Explain.
+>authentication matches login credentials to ensure that  a user is seeing their own information.
+
 * What’s the difference between authentication and authorization?
+>authorization is allowing certain users to see certain things, such as admin credentials
+
 * What’s a before filter?
+>does the action before things. such as validating a user as an admin before logging in or redirecting.
+
 * How do we keep track of a user once they’ve logged in?
+>current_user / session id 
 * When do you want to namespace a resource? When do you want to nest a resource? What's the differences between those two approaches?
+>you want to namespace when you want to authorize some users to do certain things and not others. Nest resource when you want to allow a user to own certain things for that user. namespace allows for someone to see all *and* acess their own, and have access to their own information. nesting allows for a user to see their specific information.
```

Comment:
One note: namespace really just changes the route. We use it to implement some admin functionality, but those aren't inherently connected.

Nested resources don't have to be about users seeing their **own** things. Can be about any relationship where one thing belongs to another.